### ISSUE TITLE
alternative html implementation

### ISSUE THREAD

**Issue by matthewmueller** [makojs/html#1]
jsdom is notoriously slow. i had this hanging around, sorry i  should have shared this earlier:

```js
// modules
let parse5 = require('parse5-utils')
let extract = require('deps-html')

// mako
mako.dependencies('html', parse)

function parse (file, build) {
    let ast = parse5.parse(file.contents.toString())
    let deps = extract(ast)
    let tree = build.tree

    deps = deps
      .filter(dep => dep.path)
      .map(dep => dependency(dep.path))

    function dependency (rel) {
      let dep = rel[0] === '/'
        ? path.join(file.cwd, rel)
        : path.resolve(file.dirname, rel)
      let depFile = tree.findFile(dep)
      if (!depFile) depFile = tree.addFile(dep)
      file.addDependency(depFile)
      return depFile
    }
  }
```

Seems to cut the time in half

**Comment by dominicbarnes** [comment]
That's awesome! I had noticed that JSDOM was really slow, I even had to raise a timeout on some other batch of tests to accommodate it.

**Comment by dominicbarnes** [comment]
I've released this with `v0.1.0`, let me know if you run into any issues!

**Comment by dominicbarnes** [comment]
I'm really excited about this change, I was planning on adding support for other things like images down the road. :)
